.github/workflows/test.yml

```diff
@@ -0,0 +1,58 @@
+name: test examples
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+  schedule:
+    # Twice a month
+    - cron: '0 0 1,15 * *'
+  workflow_dispatch:
+
+jobs:
+
+  # Collect all relative paths containing a 'snap/snapcraft.yaml'
+  collect_dirs:
+    runs-on: ubuntu-latest
+    outputs:
+      dirs: ${{ steps.dirs.outputs.dirs }}
+    steps:
+      - uses: actions/checkout@v3
+      - id: dirs
+        run: echo "dirs=$(find . -type f -name 'snapcraft.yaml' | sed -r 's|\/snap/[^/]+$||' |sort |uniq |jq --raw-input --slurp --compact-output 'split("\n")[:-1]')" >> $GITHUB_OUTPUT
```

Comment:
Would strict regex need `[^\/]` instead of `[^/]` for the sed match?

Reply:
Unfortunately is isn't as straightforward. Note that we could probably go with something as simple as `s|snap/snapcraft.yaml||` instead.